services/tns_submission_queue/tns_submission_queue.py

```diff
@@ -306,33 +338,66 @@ def build_reporters_string(submission_request, source, tnsrobot, session):
                 f'No authors found for tnsrobot {tnsrobot_id} and source {obj_id}, cannot report to TNS.'
             )
+
+        # if any of the users are missing first/last names, we don't submit
+        authors_with_missing_names = [
+            author
+            for author in authors
+            if author.first_name in [None, ""] or author.last_name in [None, ""]
+        ]
+        if len(authors_with_missing_names) > 0:
+            raise TNSReportError(
+                f'One or more authors are missing a first or last name: {", ".join([author.username for author in authors_with_missing_names])}, cannot report {obj_id} to TNS.'
+            )
         # if any of the users are missing an affiliation, we don't submit
         authors_with_missing_affiliations = [
-            author for author in authors if len(author.affiliations) == 0
+            author
+            for author in authors
+            if len(author.affiliations) == 0
+            or all([affiliation in [None, ""] for affiliation in author.affiliations])
         ]
         if len(authors_with_missing_affiliations) > 0:
             raise TNSReportError(
                 f'One or more authors are missing an affiliation: {", ".join([author.username for author in authors_with_missing_affiliations])}, cannot report {obj_id} to TNS.'
             )
 
-        reporters = ', '.join(
-            [
-                f'{author.first_name} {author.last_name} ({author.affiliations[0]})'
-                for author in authors
+        reporters = []
+        for author in authors:
+            # filter out empty affiliations
+            affiliations = [
+                affiliation
+                for affiliation in author.affiliations
+                if affiliation not in [None, ""]
+            ]
+            # we already verified that there is at least one affiliation earlier, so no need to check here
+            #
+            # capitalize the first letter of each affiliation
+            affiliations = [
+                affiliation[0].upper() + affiliation[1:]
+                if len(affiliation) > 1
+                else affiliation[0].upper()
+                for affiliation in affiliations
             ]
-        )
-        if tnsrobot.acknowledgments in [None, ""]:
-            raise TNSReportError(
-                f'No acknowledgments found for tnsrobot {tnsrobot_id}, cannot report {obj_id} to TNS.'
-            )
 
-        reporters += f' {str(tnsrobot.acknowledgments).strip()}'
+            # sort alphabetically (A -> Z) to ensure consistent ordering
+            affiliations = sorted(affiliations)
+
+            affiliations = ", ".join(affiliations)
+            reporters.append(f'{author.first_name} {author.last_name} ({affiliations})')
+
+        reporters = ", ".join(reporters)
+
+        # acknowledgments are added to the end of the reporters string if they exist (optional)
+        if tnsrobot.acknowledgments not in [None, ""]:
```

Comment:
This should be ' ' or '\n'?

Reply:
You mean on the join? Should be a comma and a space, there is no new line on TNS reports as far as I know

Reply:
Ah maybe you meant the warning that's a few lines under that. If so, just fixed it